test/test_util.h

```diff
@@ -728,6 +728,43 @@ std::ostream& operator<<(std::ostream& os, const CUB_NS_QUALIFIER::KeyValuePair<
     return os;
 }
 
+#if CUB_IS_INT128_ENABLED
+static std::ostream& operator<<(std::ostream& os, __uint128_t val)
+{
+  constexpr int max_digits = 40;
+  char buffer[max_digits] = {};
+  char* digit = buffer + max_digits - 1;
+  const char* ascii = "0123456789";
+
+  do 
+  {
+    *digit = ascii[val % 10];
+    val /= 10;
+    digit--;
+  }
+  while(val != 0);
+
+  for (; digit != buffer + max_digits; digit++) {
```

Comment:
I think that at the beginning of this loop, `digit` points to a position where no digit has been written.

Reply:
Nice catch, thanks!